.github/workflows/test.yml

```
@@ -0,0 +1,143 @@
+name: test
+
+on:
+  pull_request:
+    paths-ignore:
+      - "**/*.asciidoc"
+      - "**/*.md"
+      - "**/*.png"
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - "**/*.asciidoc"
+      - "**/*.md"
+      - "**/*.png"
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: static-checks-unit-tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version:
+          - "7.2"
+          - "7.3"
+          - "7.4"
+          - "8.0"
+          - "8.1"
+        dockerfile:
+          - "Dockerfile"
+          - "Dockerfile.alpine"
+    env:
+      PHP_VERSION: ${{ matrix.php-version }}
+      DOCKERFILE: ${{ matrix.dockerfile }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Prepare
+        run: make -f .ci/Makefile prepare
+      - name: Build
+        run: make -f .ci/Makefile build
+      - name: Static Check / Unit tests
+        run: make -f .ci/Makefile static-check-unit-test
+      - name: Build parts for packages
+        run: make -f .ci/Makefile generate-for-package
+      - uses: actions/upload-artifact@v3
+        with:
+          name: package-parts
+          path: src/ext/modules/*.so
+  build-packages:
+    runs-on: ubuntu-latest
+    needs:
+      - test
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: package-parts
+          path: src/ext/modules
+      - name: package
+        run: make -C packaging package
+      - name: package info
+        run: make -C packaging info
+      - uses: actions/upload-artifact@v3
+        with:
+          name: package
+          path: |
+            build/packages/*
+            !build/packages/**/*.sha512
+  test-packages-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      include: ${{ steps.generate.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: generate
+        run: |
+          MATRIX=$(.ci/generate_package_lifecycle_test_matrix.sh agent-upgrade | jq --raw-input --slurp -c 'split("\n") | map(select(length > 0)) | map(split(",")) | map({ "item": . } )')
+          echo "matrix=${MATRIX}" >> $GITHUB_OUTPUT
+  test-packages:
+    needs:
+      - build-packages
+      - test-packages-matrix
+      - test
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 10
```

I guess, `10` is likely too low in this particular case, as Sergey might wanna run more things in parallel, let's try 2x more, if possible 

```suggestion
      max-parallel: 20
```

---

https://github.com/elastic/apm-agent-php/pull/863/commits/88113890cfbd6977e17c45356906dbabe39c9258